# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1315767905 14400
#      Sun Sep 11 15:05:05 2011 -0400
# Node ID 9207cdd7ff405264832878180139f1bb886a3c62
# Parent  d7f74a8bdc46281ec0e07b980639b82a59b43afc
plot/axis.m: Partial fix for Bug #34232.

diff --git a/scripts/plot/axis.m b/scripts/plot/axis.m
--- a/scripts/plot/axis.m
+++ b/scripts/plot/axis.m
@@ -314,17 +314,17 @@ function lims = __get_tight_lims__ (ca, 
   hg_kids = findobj (ca, "-property", strcat (ax, "data"), "type", "hggroup");
   kids = setdiff (kids, hg_kids);
   if (isempty (kids))
     ## Return the current limits.
     lims = get (ca, strcat (ax, "lim"));
   else
     data = get (kids, strcat (ax, "data"));
     scale = get (ca, strcat (ax, "scale"));
-    if (strcmp (scale, "log"))
+    if (strcmp (scale, "log") && any (data > 0))
       data(data<=0) = NaN;
     end
     if (iscell (data))
       data = data (find (! cellfun ("isempty", data)));
       if (! isempty (data))
         lims_min = min (cellfun ("min", cellfun ("min", data, 'uniformoutput', false)(:)));
         lims_max = max (cellfun ("max", cellfun ("max", data, 'uniformoutput', false)(:)));
         lims = [lims_min, lims_max];
